src/beacon-api/app/crud/crud_diagnostics.py

```
@@ -0,0 +1,573 @@
+            db_obj.description = obj_in.description
+
+        if obj_in.vendor_id is not None:
+            if not obj_in.vendor_id:
+                db_obj.vendor_id = None
+            elif isinstance(obj_in.vendor_id, UUID):
+                db_obj.vendor_id = obj_in.vendor_id
+            else:
+                try:
+                    db_obj.vendor_id = UUID(str(obj_in.vendor_id))
+                except (ValueError, TypeError):
+                    v = db.query(Vendor).filter(Vendor.name.ilike(str(obj_in.vendor_id))).first()
```

Ruff (0.16.3)

[error] Undefined name `Vendor`

(F821)